src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('throws an error');
+    } else if (a + b <= c || c + b <= a || c + a <= b) {
+      throw new Error(`
+        throws an error: sides ${a}, ${b} and ${c} can’t form a triangle
+      `);
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape = 'circle';
 
-export class Triangle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('throws an error');
+    }
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
 
-export class Rectangle implements Figure {}
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape = 'rectangle';
+
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('throws an error');
+    }
```

Comment:
you can create function like

```
function checkOnError(errorMessage: string, ...args: number[]): void {
  if (args.some((arg: number) => arg <= 0)) {
    throw new Error(errorMessage);
  }
}
```
It is needed because you have the same logic in three places.